### PR TITLE
Add an ability to match machine configuration to OpenShift container by annotations

### PR DESCRIPTION
**Pull request by akorneta**
### What does this PR do?
Provides a way to implement converter from docker image recipes into OpenShift recipes.
Provides an ability to use simple (not composite) machine names in OpenShift recipes.
Fix provisioning of Che debug property in deploy_che.sh.
Adds logs filtering for foreign containers while workspace start.

### What issues does this PR fix or reference?
#7308 
#7367 

#### Release Notes
n/a

### CODE REVIEW

**Review by riuvshin — APPROVED** [review]

File: infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/Names.java [hunk 1]
```diff
@@ -20,40 +25,37 @@
...
   /**
-   * Returns pod name that is extracted from machine name.
-   *
-   * @see #machineName(String, String)
+   * Returns machine name that has the following format `{POD_NAME}/{CONTAINER_NAME}` or machine
+   * name from configuration when it's not composite.
    */
-  public static String podName(String machineName) {
-    return machineName.split("/")[0];
+  public static String machineName(Pod pod, Container container) {
```
Comment (mshaposhnik) [inline infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/Names.java]: You dont use anything from container except the name. Maybe leave param `containerName`?

Comment (sleshchenko) [reply infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/Names.java]: I'd leave `Container` as it gives us an opportunity to use something else from container later.

File: infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftInternalRuntimeTest.java [hunk 2]
```diff
@@ -121,22 +130,22 @@
...
 
-  private OpenShiftInternalRuntime internalRuntime;
+  private OpenShiftInternalRuntime internalRuntimeSpy;
 
   @BeforeMethod
   public void setup() throws Exception {
     MockitoAnnotations.initMocks(this);
-    internalRuntime =
-        new OpenShiftInternalRuntime(
+    internalRuntimeSpy =
```
Comment (sleshchenko) [inline infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftInternalRuntimeTest.java]: Are you sure that you need spy here?

File: infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/Names.java [hunk 3]
```diff
@@ -20,40 +25,37 @@
...
-  /** Returns machine name that has the following format `{POD_NAME}/{CONTAINER_NAME}`. */
-  public static String machineName(Pod pod, Container container) {
-    return machineName(pod.getMetadata().getName(), container.getName());
-  }
-
   /**
-   * Returns pod name that is extracted from machine name.
-   *
-   * @see #machineName(String, String)
+   * Returns machine name that has the following format `{POD_NAME}/{CONTAINER_NAME}` or machine
```
Comment (sleshchenko) [inline infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/Names.java]: Please rewrite java doc here to make clear that there are three possible ways in which machine name can be constructed. 
I mean
1. MACHINE_NAME_ANNOTATION_FMT
2. CHE_ORIGINAL_NAME_LABEL
3. Names of objects

**Review by sleshchenko — APPROVED** [review]
Looks good to me. A special thank for repairing che debug mode! :+1:

**Review by garagatyi — APPROVED** [review]

**Pull request merged**